### ISSUE TITLE
SSH Session Disconnect Taking Long Time After Running TMOS Command!

### ISSUE THREAD

**Issue by sindhujit1** [ktbyers/netmiko#3290]
I am trying to use netmiko to run a command on TMOS. Basically the command checks if routing domain is enabled on the F5s.

```
ssh = netmiko.ConnectHandler(device_type="f5_ltm", host="x.x.x.x", username=username_lb,
											 password=password_lb,
											 port=22, verbose=False,session_timeout=180,conn_timeout=120,timeout=120,auth_timeout=30)
ssh.disable_paging()
output = {}
output['Change Output'] = ssh.send_command('run util zebos check', delay_factor=2)
print(output['Change Output'], "output['Change Output']")

ssh.disconnect()
print('Connection closed.')
```

The code runs fine till it reaches ssh.disconnect() and I think it reaches its timeout 2 mins and then exits. 
Is this happening because we are not into ZebOS on the F5s and running the command from TMOS, so its not real CLI commands , and so ssh.disconnect() does not work ?

Trying to understand why this is behaving this way. Any help is appreciated.

**Comment by ktbyers** [comment]
@sindhujit1 Which version of Netmiko are you using?

So everything is fine until you try to do `ssh.disconnect()`.

On your device, what does the logout/exit process look like for SSH? (when you exit manually from the CLI).

**Comment by sindhujit1** [comment]
netmiko==3.4.0

Yes everything works until I do ssh.disconnect()


[root@xxxx:Standby:Disconnected] config # exit 
logout

It logs us out from the device.

Cisco nexus devices does not have this issue, so wondering if it's something specific to F5s.

**Comment by ktbyers** [comment]
@sindhujit1 You probably need to upgrade to Netmiko 4.2.0 and re-test there (i.e. Netmiko 3.4.0 is pretty old and any fix/solution will be based on Netmiko 4.2.0 code).

**Comment by TonsenWei** [comment]
I found a workaround(I'm using linux device):
it take 10 seconds in netmiko/linux/linux_ssh.py, line 106
```   
 def exit_enable_mode(self, exit_command: str = "exit") -> str:
        """Exit enable mode."""
        output = ""
        if self.check_enable_mode():
            self.write_channel(self.normalize_cmd(exit_command))
            output += self.read_until_pattern(pattern=exit_command)
            output += self.read_until_pattern(pattern=self.prompt_pattern)
            # Nature of prompt might change with the privilege deescalation
            self.set_base_prompt(pattern=self.prompt_pattern)
            if self.check_enable_mode():
                raise ValueError("Failed to exit enable mode.")
        return output
```
I modify :
```
output += self.read_until_pattern(pattern=self.prompt_pattern)
```
to:
```
output += self.read_until_pattern(pattern=self.prompt_pattern, read_timeout=1)
```

just add read_timeout=1, then it will disconnect in 1 seconds. 
It works but I must modify the source code in local. Hopefully the maintainer can improve this code.
